### DISCUSSION TITLE
wip/ideas

### DISCUSSION THREAD

**Discussion by kyx0r** [kyx0r/nextvi#137] [General]
This is Why We Can Have Nice Things

**Comment by kyx0r** [comment]
Next on my plate:
Direct syntax highlight injection support:
rewrite how hlp is implemented, do not use regex since positions already known

**Reply by LUSEDOU** [reply]
What is the state of this?
I would like to switch to nextvi. But lsp, highlight is a wall. So i would like to help in this but do not how

**Reply by kyx0r** [reply]
It is completed. You can inject the highlighting directly by populating led_attsb which holds led_att structures.
For example, see how hlp option is implemented now:
https://github.com/kyx0r/nextvi/blob/master/vi.c#L1829-L1848

This code is all you need to implement self contained syntax highlighting that can use custom algorithms outside of nextvi regex engine.
This is quite efficient since highlights are terminal size bound, so allocating led_att per screen character is not so bad.

It also works flawlessly when text is reordered and rendered in reverse directions which is what makes is so hard to do.

**Reply by kyx0r** [reply]
If you want to see hlp in action, open .c file. For example vi.c and run :sehlp

**Reply by LUSEDOU** [reply]
Oh, yes. I see the conf.c file
So i just have to define the keywords and their color for zig, dont I? Ok

Theres anything else in the roadmap?

**Reply by kyx0r** [reply]
Indeed. conf.c will be sufficient for most use cases.
As far as the road-map goes, I don't have any grand ideas. Just improving core flexibility & architecture while keeping number of lines of code < 7k. Most recent developments on master have been focused on providing means of improving long line performance. New ex options like lim and seq.

More advanced features are being implemented on patches branch. If you like to help out, soft line wrap patch still needs some work (fixing certain edge cases with undo/redo, editing lines, etc).

https://github.com/kyx0r/nextvi/tree/patches

**Comment by kyx0r** [comment]
I was thinking (but never really got to) about implementing a patch for searching (fuzzing) as you type. So for example user does / and starts typing and editor searches immediately and moves the screen to closest match.
